src/promptflow-devkit/promptflow/_sdk/entities/_chat_group/_chat_role.py

```diff
@@ -28,10 +30,33 @@ class ChatRole:
     :type inputs: Optional[Dict]
     """
 
-    def __init__(self, flow: Union[str, PathLike], role: str, inputs: Optional[Dict] = None, **kwargs):
+    def __init__(self,
+                 flow: Union[str, PathLike],
+                 role: str,
+                 inputs: Optional[Dict] = None,
+                 name: Optional[str] = None,
+                 stop_signal: Optional[str] = None,
+                 flow_file: Optional[Path] = None,
+                 working_dir: Optional[Path] = None,
+                 connections: Optional[Dict[str, Any]] = None,
+                 inputs_mapping: Optional[Dict[str, str]] = None,
+                 **kwargs):
         self._role = role
         self._flow, self._flow_object = self._validate_flow(flow)
-        self._inputs, self._outputs = self._build_role_io(flow, inputs)
+        if inputs is not None:
```

Comment:
The condition should be `if flow is not None`, `inputs` is just the value, the function will parse the flow dag to get inputs meta anyways.

Reply:
BTW could we just use `self.flow` to replace `self._flow_file`?

Reply:
Update to reuse self.flow. For cloud batch run, I don't assign value to `inputs`. So I check whether it is now before `build_role_io`

Reply:
Per offline synced, put inputs checking in `_build_role_io`